packages/api/src/auth.js

```diff
@@ -36,8 +36,8 @@ class Auth {
     return querystring.parse(respText);
   }
 
-  async retrieveUserData (accessToken) {
-    const result = await this.doFetch('https://api.github.com/user', {
+  async retrieveRepoData (accessToken) {
+    const result = await this.doFetch('https://api.github.com/user/repos', {
```

Comment:
This is querying all repos for which the user has an explicit role: https://docs.github.com/en/rest/reference/repos#list-repositories-for-the-authenticated-user

All auth checks are based on this data. I'm a bit concerned it's not quite right, though, because when I ran it for me it didn't include flaky.dev. An alternative would be to query all data about the repo for which the action is being requested, but that requires knowing the repo's owner: https://docs.github.com/en/rest/reference/repos#check-if-a-user-is-a-repository-collaborator

Reply:
@cedpeters there's an API method that gives you the permission level on a given repository for a given user:

https://docs.github.com/en/rest/reference/repos#check-if-a-user-is-a-repository-collaborator

I think we could use this method, and confirm that the username returned via the OAuth dance.

Alternatively, we could use:

https://docs.github.com/en/rest/reference/repos#get-repository-permissions-for-a-user

Which gives us the actual permissions for a single user ... this might be better as it gives us a bit more granularity, letting us know if the user is an admin vs., a collaborator.

Reply:
The issue with those is that the path requires the owner of the repo. For example, for repo permissions for a user, the path is: 
`/repos/{owner}/{repo}/collaborators/{username}/permission`
The UI doesn't know the owner of a repo by default, and it's not currently part of the DELETE /repo endpoint. Unless the owner is just the org?